app/Domain/Integrations/Rules/UniqueIntegrationUrl.php

```diff
@@ -36,7 +40,9 @@ public function validate(string $attribute, mixed $value, Closure $fail): void
                 continue;
             }
 
-            $fail('validation.distinct')->translate();
+            if ($inUse[$hash] === $key) {
+                $fail('validation.distinct')->translate();
+            }
```

Comment:
```suggestion
            if ($inUse[$hash] !== $key) {
                continue;
            }
            
            $fail('validation.distinct')->translate();
```

Can you make this a early retrun aswell?